datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -607,6 +656,15 @@ def _submit_metric(
         if hostname is None:
             hostname = ''
 
+        if (name in self.exclude_metrics_filters) or (
+            self.exclude_metrics_pattern is not None and self.exclude_metrics_pattern.search(name)
+        ):
+            return
+        elif (self.include_metrics_filters and name not in self.include_metrics_filters) or (
+            self.include_metrics_pattern is not None and not self.include_metrics_pattern.search(name)
+        ):
```

Comment:
I may be missing something, but the logic here seems odd. If I have:
```
include_metrics_filters:
 - foo
 - ^bar
```

Neither `include_metrics_filters` nor `include_metrics_pattern` are None, so for neither `foo` (does not match patterns) nor `bar` (not in the list of filters) would be included, but I think both should?

Reply:
Please do:

```python
    def should_send_metric(self, metric_name):
        return not self._metric_excluded(metric_name) and self._metric_included(metric_name)

    def _metric_included(self, metric_name):
        if self.include_metrics_pattern is None:
            return True

        return self.include_metrics_pattern.search(metric_name) is not None

    def _metric_excluded(self, metric_name):
        if self.exclude_metrics_pattern is None:
            return False

        return self.exclude_metrics_pattern.search(metric_name) is not None
```

Reply:
Thanks for the suggestion, it's much cleaner now!